pkgs/development/beam-modules/credo/default.nix

```
@@ -0,0 +1,56 @@
+{ lib
+, fetchFromGitHub
+, makeWrapper
+, beamPackages
+, elixir
+, erlang
+, rebar3
+, ...
+}:
+
+beamPackages.mixRelease rec {
+  pname = "credo";
+  version = "1.7.4";
+
+  inherit elixir;
+
+  buildInputs = [ erlang ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  src = fetchFromGitHub {
+    owner = "rrrene";
+    repo = "credo";
+    rev = "v${version}";
+    hash = "sha256-C3/h0nc0r1CWxRwXIccFjlIRdOoh9NyDTV62VZgUU7k=";
+  };
+
+  patches = [ ./01-enable-ansi.diff ];
+
+  mixFodDeps = beamPackages.fetchMixDeps {
+    pname = "mix-deps-${pname}";
+    inherit src version;
+    sha256 = "sha256-nhTmH5FkZDVbjIVtYSfbQJ44fu/LRLLbGwEE3EriSTs=";
+  };
+
+  installPhase = ''
+    mix escript.build
+
+    mkdir -p $out/bin
+    mv ./credo $out/bin
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall
    mix escript.build

    mkdir -p $out/bin
    mv ./credo $out/bin
    runHook postInstall
  '';
```

when overriding phases you must run their hooks manually.

---

Done, thank you!